wadsrc/static/zscript/menu/optionmenu.txt

```diff
@@ -536,3 +536,75 @@ class CompatibilityMenu : OptionMenu
 			DTA_CleanNoMove_1, true);
 	}
 }
+
+class GLTextureGLOptions : OptionMenu
+{
+	private int mWarningIndex;
+
+	override void Init(Menu parent, OptionMenuDescriptor desc)
+	{
+		super.Init(parent, desc);
+
+		for (int i=0; i < mDesc.mItems.Size(); ++i)
+		{
+			if (mDesc.mItems[i].mLabel == "!HQRESIZE_WARNING!")
+			{
+				mWarningIndex = i;
+				break;
+			}
+		}
+	}
+
+	override void Ticker()
+	{
+		Super.Ticker();
+
+		if (mWarningIndex > 0)
+		{
+			string message;
+
+			if (gl_texture_hqresize > 0)
+			{
+				int multiplier;
+
+				switch (gl_texture_hqresize)
```

Comment:
New modes need to be handled in this switch to display correct information about memory usage.